forge/test/mlir/test_ops.py

```diff
@@ -633,3 +633,54 @@ def forward(self, input_tensor, indices):
     # Validate results
     assert compare_with_golden_pcc(golden=golden_out.value(), calculated=co_out[0], pcc=0.99)
 
+
+@pytest.mark.parametrize("input_shape", [
+    (2, 32, 64, 64), 
+    (3, 22, 37, 41),
+    (2, 32, 64),
+    (3, 22, 37),
+], ids=["(2, 32, 64, 64)", "(2, 22, 37, 41)", "(2, 32, 64)", "(3, 22, 37)"])
+@pytest.mark.parametrize("dim", [
+    0,
+    1,
+    2,
+    3,
+    -1, 
+    -2,
+    -3,
+    -4,
+], ids=["0", "1", "2", "3", "-1", "-2", "-3", "-4"])
+def test_reduce_max(input_shape, dim):
+    
+    reduce_max_dim = dim
+    if reduce_max_dim < 0:
+        reduce_max_dim = reduce_max_dim + len(input_shape)
+    if (reduce_max_dim < 0) or (reduce_max_dim >= len(input_shape)):
+        pytest.skip()
+        
+    if (input_shape in [(2, 32, 64, 64), (3, 22, 37, 41)] and dim in [0, -4, 1, -3]) or (input_shape in [(2, 32, 64), (3, 22, 37)] and dim in [0, -3]):
+        pytest.xfail("TTNN Issue: Unsupported dim")
+    
+    # TTNN Max issues:
+    #   Unsupported dim - https://github.com/tenstorrent/tt-metal/issues/13186
+    #   Shape mismatch along the H and W dimension - https://github.com/tenstorrent/tt-metal/issues/13189
+    #   Tensor rank is not 4 - https://github.com/tenstorrent/tt-metal/issues/13190
+
```

Comment:
@nvukobratTT correct me if i'm wrong, but i think we should add these issues to `TT-Forge` project as well (in `External ttnn/tt-metal` component) - i don't see them there: https://github.com/orgs/tenstorrent/projects/54/views/8?sliceBy%5Bvalue%5D=External+ttnn%2Ftt-metal

Reply:
Yup!

@chandrasekaranpradeep I added those to TT-Forge project + assigned P1 as label. Does one of these combinations exist on ResNet? If yes, please let me know and we'll raise it to P0.

Reply:
@nvukobratTT @pilkicTT Thanks for adding  those issues to TT-Forge Project.
@nvukobratTT Regarding label change from P1 to P0, I have tested the max_pool_2d op with exact configuration(i.e input_shape, kernel_size, stride, padding) from the Resnet50 model (i.e [forge/test/mlir/resnet/test_resnet_inference.py](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/resnet/test_resnet_inference.py)) for finding the input_shape and dim for the reduce_max op support but the [currect max_pool_2d](https://github.com/tenstorrent/tt-forge-fe/blob/a733ab689e0839cadad66765132ddbe87c77ea92/forge/forge/op/eval/forge/pooling.py#L25) is not decomposed to reduce_max op, it is directly mapped to the TTIR. We can have the P1 label  for reduce_max op issues raised in tt-metal.